File: CPP [3501-4000]/3502.minimum-cost-to-reach-every-position [1].cpp
class Solution {
public:
    // T.C.=O(n), S.C.=O(1)
    vector<int> minCosts(vector<int>& cost) {
        vector<int> ans;
        int minCost = 100;
        
        for(int currCost: cost)
        {
            minCost = min(minCost, currCost);
            ans.push_back(minCost);
        }

        return ans;
    }
};

File: Javascript/3502.minimum-cost-to-reach-every-position [1].js
/**
 * @param {number[]} cost
 * @return {number[]}
 */
var minCosts = function(cost) {
    const ans = []
    let minCost = 100
    
    for(const currCost of cost){
        minCost = Math.min(minCost, currCost)
        ans.push(minCost)
    }

    return ans
};